clickhouse_connect/driver/client.py

```diff
@@ -47,7 +47,7 @@ def _apply_settings(self, settings: Dict[str, Any] = None):
     def _validate_settings(self, settings: Dict[str, Any]):
         validated = {}
         for key, value in settings.items():
-            if 'session' not in key:
+            if 'session' not in key and key not in ('database', 'buffer_size'):
```

Comment:
So 'database', 'buffer_size' and keys with 'session' could be missing or readonly and then we set them regardless?
Maybe check that they are not readonly? Or at least add a comment explaining why it is safe to override those readonly settings?

Reply:
database and buffer_size are not actual settings, they can always be sent as query parameters with any HTTP request, so they aren't ever "read only".  I wanted to make sure they weren't stripped from the request by the logic that checks valid settings.

Reply:
Same with the session parameters "session_id" and "session_timeout"